framework/src/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -1202,6 +1204,36 @@ public static FlowExpressionContext buildContextForMethodUse(
             return flowExprContext;
         }
 
+        /**
+         * @return a {@link FlowExpressionContext} for the method {@code methodInvocation}
+         *     (represented as a {@link MethodInvocationTree} as seen at the method use (i.e., at a
+         *     method call site).
```

Comment:
Would cross-references between the Node and Tree versions of this method be useful?

Reply:
I don't think so.

Reply:
Why is the implementation of `buildContextForMethodUse(MethodInvocationNode methodInvocation, BaseContext checkerContext)` above not simply

```
    return buildContextForMethodUse(methodInvocation.getTree(), checkerContext);
```
?

Are the two methods calculating slightly different things? Then that should be documented.
Is this duplication more efficient? Then please document it.
If the duplication isn't intended, let's remove it.

Reply:
methodInvocation.getTree() might return null.   Every single method that creates a `FlowExpressionContext` has one version for Trees and one for Nodes.

Reply:
What methods do you mean? For example for the similar `buildContextForNewClassUse` I only see a version that takes a Node.
Wouldn't it be nice to remove that duplication?

Reply:
Again, we can't remove the duplication because node.getTree() might return null.  So, we need both methods.

Reply:
Ok, but which methods do you mean when you say `Every single method that creates a FlowExpressionContext has one version for Trees and one for Nodes.`?
I'm trying to understand whether this duplication is already happening or not.

Reply:
I see the other methods all use trees, except for `buildContextForNewClassUse`.   I was thinking of the `internalReprOf` methods in FlowExpressions.